main/src/main/java/org/mobilitydata/gtfsvalidator/runner/ValidationRunner.java

```diff
@@ -0,0 +1,237 @@
+  private static final String GTFS_ZIP_FILENAME = "gtfs.zip";
+
+  public enum Status {
+    // Indicates validation successfully completed, but doesn't imply the
+    // feed itself is valid.
+    SUCCESS,
+    // Indicates validation did not successfully complete, with exceptions
+    // caught and written to the system errors JSON output.
+    SYSTEM_ERRORS,
+    // Indicates validation did not successfully complete, with exceptions
+    // caught and written only to console logging.
+    EXCEPTION
+  }
+
+  public Status run(ValidationRunnerConfig config) {
+    ValidatorLoader validatorLoader = null;
+    try {
+      validatorLoader = new ValidatorLoader();
+    } catch (ValidatorLoaderException e) {
+      logger.atSevere().withCause(e).log("Cannot load validator classes");
+      return Status.EXCEPTION;
+    }
+    GtfsFeedLoader feedLoader = new GtfsFeedLoader();
+
+    logger.atInfo().log("validation config:\n%s", config);
+    logger.atInfo().log("validators:\n%s", validatorLoader.listValidators());
+
+    final long startNanos = System.nanoTime();
+    // Input.
+    feedLoader.setNumThreads(config.numThreads());
+    NoticeContainer noticeContainer = new NoticeContainer();
+    GtfsFeedContainer feedContainer;
+    GtfsInput gtfsInput = null;
+    try {
+      gtfsInput = createGtfsInput(config);
+    } catch (IOException e) {
+      logger.atSevere().withCause(e).log("Cannot load GTFS feed");
+      noticeContainer.addSystemError(new IOError(e));
+    } catch (URISyntaxException e) {
+      logger.atSevere().withCause(e).log("Syntax error in URI");
+      noticeContainer.addSystemError(new URISyntaxError(e));
+    }
+    if (gtfsInput == null) {
+      exportReport(noticeContainer, config);
+      if (!noticeContainer.getSystemErrors().isEmpty()) {
+        return Status.SYSTEM_ERRORS;
+      } else {
+        return Status.EXCEPTION;
+      }
+    }
+    ValidationContext validationContext =
+        ValidationContext.builder()
+            .setCountryCode(config.countryCode())
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+    try {
+      feedContainer =
+          loadAndValidate(
+              validatorLoader, feedLoader, noticeContainer, gtfsInput, validationContext);
+    } catch (InterruptedException e) {
+      logger.atSevere().withCause(e).log("Validation was interrupted");
+      return Status.EXCEPTION;
+    }
+    closeGtfsInput(gtfsInput, noticeContainer);
+
+    // Output
+    exportReport(noticeContainer, config);
+    printSummary(startNanos, feedContainer);
+    return Status.SUCCESS;
+  }
+
+  /**
+   * Prints validation metadata.
+   *
+   * @param startNanos start time as nanoseconds
+   * @param feedContainer the {@code GtfsFeedContainer}
+   */
+  public static void printSummary(long startNanos, GtfsFeedContainer feedContainer) {
+    final long endNanos = System.nanoTime();
+    if (!feedContainer.isParsedSuccessfully()) {
```

Comment:
The `Status` codes put us in a good position to handle https://github.com/MobilityData/gtfs-validator/issues/1096 as well via something like `Status.PARSING_FAILED`, which is to surface the condition where parsing failed and many validators weren't invoked. That can be in a future PR.